lib/core/src/display/shape/text/render.rs

```diff
@@ -0,0 +1,202 @@
+        // at once.
+        let x_margin           = (displayed_lines as f32) * line_height / space_width;
+        let max_glyphs_in_line = (displayed_chars + 2.0 * x_margin).floor() as usize + ADDITIONAL;
+        GlyphLinesAssignment::new(displayed_lines,max_glyphs_in_line,x_margin)
+    }
+}
+
+
+// === DisplayObject ===
+
+impl From<&RenderedContent> for DisplayObjectData {
+    fn from(rendered_content:&RenderedContent) -> Self {
+        rendered_content.display_object.clone_ref()
+    }
+}
+
+
+// === Update ===
+
+impl RenderedContent {
+    /// Update all displayed glyphs.
+    pub fn update_glyphs(&mut self, content:&mut TextFieldContent, fonts:&mut FontRegistry) {
+        let glyph_lines       = self.glyph_lines.iter_mut().enumerate();
+        let lines_assignment  = glyph_lines.zip(self.assignment.glyph_lines_fragments.iter());
+        let assigned_lines    = lines_assignment.filter_map(|(l,opt)| opt.as_ref().map(|f|(l,f)));
+        let dirty_lines       = std::mem::take(&mut content.dirty_lines);
+        let dirty_glyph_lines = std::mem::take(&mut self.assignment.dirty_glyph_lines);
+        for ((index,glyph_line),fragment) in assigned_lines {
+            if dirty_glyph_lines.contains(&index) || dirty_lines.is_dirty(fragment.line_index) {
+                let mut f_content = content.full_info(fonts);
+                let bsl_start     = Self::baseline_start_for_fragment(fragment,&mut f_content);
+                let line          = &content.lines[fragment.line_index];
+                let chars         = &line.chars()[fragment.chars_range.clone()];
+                glyph_line.set_baseline_start(bsl_start);
+                glyph_line.replace_text(chars.iter().cloned(),fonts);
+            }
+        }
+        self.display_object.update();
+    }
+
+    /// Update all displayed cursors with their selections.
+    pub fn update_cursors(&mut self, cursors:&Cursors, content:&mut TextFieldContentFullInfo) {
+        let cursor_system = &self.cursor_system;
+        self.cursors.resize_with(cursors.cursors.len(),|| Self::new_cursor_sprites(cursor_system));
+        for (sprites,cursor) in self.cursors.iter_mut().zip(cursors.cursors.iter()) {
+            let position = Cursor::render_position(&cursor.position,content);
+            sprites.cursor.set_position(Vector3::new(position.x,position.y,0.0));
+            sprites.cursor.size().set(Vector2::new(2.0,self.line_height));
+
+            let selection = cursor.selection_range();
+            let line_height   = self.line_height;
+            let system        = &self.selection_system;
+            let mut generator = SelectionSpritesGenerator {content,line_height,system};
+            sprites.selection.clear();
+            sprites.selection = generator.generate(&selection);
+        }
+        self.display_object.update();
+    }
+
+    fn baseline_start_for_fragment(fragment:&LineFragment, content:&mut TextFieldContentFullInfo)
```

Comment:
Needs a doc. What does it mean? Especially what does "LineFragment" mean in this context? I know I can jump here . aand there to leanr more, but small docs with explanation would be better here :)